alpha/declcfg/declcfg.go

```diff
@@ -93,16 +93,19 @@ type RelatedImage struct {
 }
 
 type Deprecation struct {
-	Schema       string             `json:"schema"`
-	Package      string             `json:"package"`
-	Name         string             `json:"name,omitempty"`
-	Deprecations []DeprecationEntry `json:"deprecations"`
+	Schema  string             `json:"schema"`
+	Package string             `json:"package"`
+	Entries []DeprecationEntry `json:"entries"`
 }
 
 type DeprecationEntry struct {
-	Schema  string          `json:"schema"`
-	Name    string          `json:"name,omitempty"`
-	Message json.RawMessage `json:"message"`
+	Reference PackageScopedReference `json:"reference"`
```

Comment:
What is the motivation for making this a separate type as opposed to keeping the flat structure?

Before this change, the YAML to represent a `Deprecation` is:
```yaml
schema: olm.deprecations
package: somepackage
entries:
  - schema: olm.package
    name: somepackage
    message: |
       somepackage is deprecated as it has been archived and is no longer maintained.
```

After the change it is now:
```yaml
schema: olm.deprecations
package: somepackage
entries:
  - reference:
     schema: olm.package
     name: somepackage
    message: |
       somepackage is deprecated as it has been archived and is no longer maintained.
```

In this context it feels unnecessary to have this nested and requires extra typing (maybe I'm just lazy?) to get the same thing conveyed.

Reply:
It felt like having the flat structure was doing some "this is really a reference" hand-waving, using a nested, repeated fieldname (schema).  Really prone to confusion, and the new structure makes it clear that this is a reference to a schema+name in a manner that loosely follows general format for a kube GVK-like thing.

Reply:
Jordan and I collab'd on this one. First it was:
```
- schema: olm.package
  name: foo
  message: foo is deprecated
```

Then Jordan had:
```
- scope: olm.package
  reference: foo
  message: foo is deprecated
```

And we settled on this because I thought it was important that `schema` and `name` show up here, but I also saw how Jordan thought that an inlined schema/name might cause confusion because it isn't an obvious reference to another object.

Reply:
I'm not opinionated enough to make this block this PR, but tried to approach it from a user's perspective. I can see the confusion but in the context of FBC objects that contain an `entries` field there isn't an explicit `reference` field because it is a reference to another object. I also think if this is well documented it mitigates _most_ of that confusion. As a user I would also prefer to type less if it accomplishes the same thing, but again I could just be a bit lazy.

Again, shouldn't block this PR but was something I thought was worth at least bringing up. Thanks for the insights!